src/OpenTelemetry.Instrumentation.AspNet/Implementation/HttpInListener.cs

```diff
@@ -61,17 +61,16 @@ public override void OnStartActivity(Activity activity, object payload)
 
             if (!(this.options.TextFormat is TraceContextFormat))
             {
-                // This requires to ignore the current activity and create a new one
-                // using the context extracted using the format TextFormat supports.
                 var ctx = this.options.TextFormat.Extract(default, request, HttpRequestHeaderValuesGetter);
-                if (ctx != default)
+
+                if (ctx.ActivityContext.IsValid() && ctx.ActivityContext != activity.Context)

```

Comment:
ASP.NET (and ASP.NET Core) already extracts W3CTraceParent and creates ActivityContext. We'll now be forced to extract it once more time by default, and compare it with the one extracted by ASP.NET. Considering this is in hot path, *before* sampling, won't we regress perf ?

Reply:
Resolving this comment, I'll respond to the one above.